src/main/scala/com/exasol/spark/DefaultSource.scala

```diff
@@ -30,12 +30,27 @@ class DefaultSource extends RelationProvider with DataSourceRegister with Schema
     schema: StructType
   ): BaseRelation = {
 
-    val (queryString, manager) = fromParameters(parameters)
+    val (queryString, manager) = fromParameters(parameters, sqlContext)
     new ExasolRelation(sqlContext, queryString, Option(schema), manager)
   }
 
+  private[spark] def mergeConfiguration(
+    parameters: Map[String, String],
+    sparkConf: Map[String, String]
+  ): Map[String, String] = {
+    var result: Map[String, String] = parameters
+    sparkConf.keySet
+      .filter(_.contains("spark.exasol"))
+      .foreach(key => {
+        val keySuffix: String = key.substring(key.lastIndexOf(".") + 1)
+        result += (keySuffix -> sparkConf.getOrElse(key, ""))
+      })
+    result
```

Comment:
I can suggest different approach here to merge two maps. For example, we can refactor similar to below:

```scala
parameters.map { case (key, value) ->
   key -> sparkConf.getOrElse(s"spark.exasol.$key", value)
}
```
Could you please check if above works?